### PR TITLE
EN-2386-PAM-show-the-task-form-in-a-new-page

### PR DESCRIPTION
**Pull request by alepintus**
Hi @joewhite101 ,
this is the cherry-pick from [this PR](https://github.com/entando/entando-components/pull/469).
Could you please have a look?
Thanks!

### CODE REVIEW

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java [hunk 1]
```diff
@@ -0,0 +1,873 @@
...
+            
+            
+            Map < String, String> toBpmStrings = toBpm.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> (String) e.getValue()));
+
+            final String result = this.getFormManager().completeHumanFormTask(config, containerId, processId, Long.valueOf(taskId), toBpmStrings);
+            logger.info("Result {} ", result);
+      
+            logger.info("TASK ID RESULT: {}", result);
+            
+            //   List<String> args = new ArrayList<>();
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java]: Remove if dead

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java [hunk 2]
```diff
@@ -0,0 +1,873 @@
...
+                        String msg = String.format("Invalid input '%s' on field '%s'", value, key);
+                        this.addFieldError(key, msg);
+                    }
+                }
+            } else {
+                logger.debug("Invalid input '{}' on field '{}'", value, key);
+                String msg = String.format("Invalid input '%s' on field '%s'", value, key);
+                this.addFieldError(key, msg);
+
+                //validationResult.add(new ApiError(IApiErrorCodes.API_VALIDATION_ERROR, msg, Response.Status.CONFLICT));
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java]: Remove if dead

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java [hunk 3]
```diff
@@ -0,0 +1,873 @@
...
+    }
+
+    
+    @Override
+    public String createNew() {
+        try {
+            logger.debug("createNew taskID {}", taskId);
+            kpfr = null;
+            setDataObjectOnSession(null);
+            //DUPLICATED CODE --> REFACTORING NEEDED
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java]: put a TODO tag on this so we can find it later

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java [hunk 4]
```diff
@@ -0,0 +1,873 @@
...
+
+        Map<String, String> map = new HashMap<>();
+        try {
+
+            inputData.keySet().forEach(f -> {
+                KieDataHolder dataHolder = null;
+                if (inputData.get(f) instanceof org.json.JSONObject) {
+                    logger.debug("  inputData.get(f) {} ", inputData.get(f));
+                    for (KieDataHolder fh : form.getHolders()) {
+                        if (fh.getValue().equals(f)) {//||(fh.getValue().equals("taskInput"+f))) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java]: Remove comment if dead

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java [hunk 5]
```diff
@@ -0,0 +1,873 @@
...
+        /*  
+            if (child.equals(holderValue) || child.equals("taskInput" + holderTypeName) || child.equals(formName)) {
+                formChild = taskData.getJSONObject(child);
+                break;
+        }*/
+
+        Map<String, String> map = new HashMap<>();
+        try {
+
+            inputData.keySet().forEach(f -> {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java]: Add a short comment here on what this is doing. This is pretty complex

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java [hunk 6]
```diff
@@ -0,0 +1,873 @@
...
+    }
+
+    public void setDataObjectModelManager(IDataObjectModelManager _dataObjectModelManager) {
+        this.dataObjectModelManager = _dataObjectModelManager;
+    }
+
+    private Map<String, String> getValuesMap(JSONObject inputData, String parentKey, KieProcessFormQueryResult form) {
+        logger.debug("**************  getValuesMap {} ", parentKey);
+        /*  
+            if (child.equals(holderValue) || child.equals("taskInput" + holderTypeName) || child.equals(formName)) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmTypeTaskFormAction.java]: Remove if dead

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is Well

**Pull request merged**